security/responding.md

```diff
@@ -0,0 +1,143 @@
+# Vulnerability Discolure Response Policy
+
+The project follows the following process when responding to vulnerabilities
+raised via the Security Disclosure Process.
+
+## Product Security Taskforce Membership
+
+The subset of maintainers who are subscribed to the security@knative.team email
+address form the Product Security Taskforce. 
+This group will be responsible for fixing vulnerabilities and managing the
+disclosure process. 
+
+The Product Security Taskforce is distinct from the embargo list: it is not
+neccessary (or helpful) to join the PST in order to recieve early notice of
+vulnerabilities. The PST is intentionally small to avoid inadvertant
+disclosures of vulnerabilities (while being large enough to spread the work of
+responding to vulnerabilities) and should be composed of those who will
+actively work on the fix and disclosure process.
+
+Anyone who is an Approver, Lead or TOC Member in a Knative Project, or who is
+an employee of a member of the Embargo list working in a security role may
+apply to be a member of this group.
+The application is approved or rejected by the TOC.
+
+## Identifying a Fix Lead
+
+The PST Fix Lead Role is responsible for initially responding to
+vulnerabilities sent to security@knative.team, and for coordinating and
+managing the response with the rest of the PST. 
+
+We use a rota system to assign the default PST Lead role. 
+All members of the PST are expected to sign up to the rota.
+Regardless of the rota, another member of the PST may volunteer to be the PST
+lead for a particular vulnerability.
+This may be appropriate, for example, when the vulnerability is in a particular
+area of the code that PST member is familiar with, or when the rota-ed PST lead
+is already handing another vulnerability.
+
+## Organising the Fix Team
+
+The Fix Lead will quickly identify relevant engineers from the affected
+projects/areas/working groups and the PST and add those engineers to the
+security disclosure / email thread. This group becomes the Fix Team for the
+vulnerability.
+
+Note: Given the current size of the project it is likely that in many cases the
+Fix Team will be the entire PST, possibly with an additional engineer or
+engineers for the affected area if the PST does not have relevant expertise
+itself.
+
+If the Fix Team decides the disclosure is in fact a vulnerability which should
+be fixed, the Fix Lead will create a Github Security Advisory and provide the
```

Comment:
Only repo admins can create a security advisory. https://docs.github.com/en/github/managing-security-vulnerabilities/about-github-security-advisories

Does this mean we make all fix leads admins? or we require PST members to be admins? Or we need to loop in an admin to create the SA? Or what?

Reply:
🤔 good question.

Reply:
I think it's reasonable to loop in a repo admin to start with once we've decided that a GHSA is needed. (Note that take-backs require going to GitHub support, so we need to be reasonably sure at this point.)

The existing repo admins will be able to see the GHSA anyway.

Reply:
How much will they be able to see? Just the extence, or the full repo and all commits, or something else? Because the more they can see, the more we are effectively adding the repo admins to the PST for that repo. And PST membership implies a certain level of trust, which is not the same as the level of trust we put in repo admins.